src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -375,25 +431,110 @@ def check_token_status(self, clustersTokenMap=None):
 
+    @Endpoint()
+    @ReadPermission
+    def security_config(self):
+        return self._get_security_config()
+
+    def _get_security_config(self):
+        orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
+        if orch_backend == 'cephadm':
+            cmd = {
+                'prefix': 'orch get-security-config',
+            }
+            ret_status, out, _ = mgr.mon_command(cmd)
+            if ret_status == 0 and out is not None:
+                security_info = json.loads(out)
+                security_enabled = security_info['security_enabled']
+                mgmt_gw_enabled = security_info['mgmt_gw_enabled']
+                return {
+                    'security_enabled': bool(security_enabled),
+                    'mgmt_gw_enabled': bool(mgmt_gw_enabled)
+                }
+        return None
+
     @Endpoint()
     @ReadPermission
     def get_prometheus_api_url(self):
+        security_content = self._get_security_config()
+        mgmt_gw_enabled = security_content['mgmt_gw_enabled']
         prometheus_url = Settings.PROMETHEUS_API_HOST
+
         if prometheus_url is not None:
-            # check if is url is already in IP format
-            pattern = r'^(?:https?|http):\/\/(?:\d{1,3}\.){3}\d{1,3}:\d+$'
-            valid_ip_url = bool(re.match(pattern, prometheus_url))
-            if not valid_ip_url:
-                parsed_url = urlparse(prometheus_url)
-                hostname = parsed_url.hostname
-                orch = OrchClient.instance()
-                inventory_hosts = [host.to_json() for host in orch.hosts.list()]
-                for host in inventory_hosts:
-                    if host['hostname'] == hostname or host['hostname'] in hostname:
-                        node_ip = host['addr']
-                prometheus_url = prometheus_url.replace(hostname, node_ip)
+            if '.ceph-dashboard' in prometheus_url:
+                prometheus_url = prometheus_url.replace('.ceph-dashboard', '')
+            parsed_url = urlparse(prometheus_url)
+            hostname = parsed_url.hostname
+            try:
+                # Check if the hostname is already an IP address
+                ipaddress.ip_address(hostname)
+                valid_ip_url = True
+            except ValueError:
+                valid_ip_url = False
+
+            if mgmt_gw_enabled:
+                # If management gateway is enabled and hostname is not an IP
+                if not valid_ip_url:
+                    orch = OrchClient.instance()
+                    inventory_hosts = [host.to_json() for host in orch.hosts.list()]
+                    for host in inventory_hosts:
+                        if host['hostname'] == hostname or hostname in host['hostname']:
+                            node_ip = host['addr']
+                            prometheus_url = node_ip  # we need just IP if mgmt gw is enabled
+                            break
```

Comment:
this code is repeated
```suggestion
                if not valid_ip_url:
                    orch = OrchClient.instance()
                    inventory_hosts = [host.to_json() for host in orch.hosts.list()]
                    for host in inventory_hosts:
                        if host['hostname'] == hostname or hostname in host['hostname']:
                            node_ip = host['addr']
                            prometheus_url = node_ip  # we need just IP if mgmt gw is enabled
                            break
```

Reply:
Updated. Thanks